packages/expo-av/android/src/main/java/expo/modules/av/AVManager.java

```diff
@@ -181,10 +184,26 @@ public void onHostPause() {
 
   @Override
   public void onHostDestroy() {
-    mContext.unregisterReceiver(mNoisyAudioStreamReceiver);
-    for (final Integer key : mSoundMap.keySet()) {
-      removeSoundForKey(key);
+    if (isRegistered) {
+      try {
+        mContext.unregisterReceiver(mNoisyAudioStreamReceiver);
+        isRegistered = false;
+      } catch (final Exception exception) {
+        // Ignore it
+      }
     }
+
+    // removeSoundForKey() on all remaining sounds
+    Iterator<PlayerData> iter = mSoundMap.values().iterator();
+    while(iter.hasNext()) {
+      final PlayerData data = iter.next();
+      iter.remove();
+      if (data != null) {
+        data.release();
+        abandonAudioFocusIfUnused();
```

Comment:
Can't we just call it once, after iteration?

Reply:
Actually, I see that abandonAudioFocus() is always called at the end of onHostDestroy(), so I think there is no need to call abandonAudioFocusIfUnused() at all. Removed.